### PR TITLE
No target name unit and e2e tests

### LINKED ISSUE #350: We need tests for target secret name defaults
We now have the target secret name defaulting to the ExternalSecret metadata name, but we need to to have a test for that.

### PR DESCRIPTION
**Pull request by omateusp**
Tests to check if use external secret name if target secret name isn't defined

Close #350

### CODE REVIEW

**Comment by knelasevero** [comment]
You are gonna need to change


			if tc.checkSecret != nil {
				syncedSecret := &v1.Secret{}
				secretLookupKey := types.NamespacedName{
					Name:      ExternalSecretTargetSecretName,
					Namespace: ExternalSecretNamespace,
				}
				
to


			if tc.checkSecret != nil {
				syncedSecret := &v1.Secret{}
				secretLookupKey = types.NamespacedName{
					Name:      ExternalSecretTargetSecretName,
					Namespace: ExternalSecretNamespace,
				}
				if createdES.Spec.Target.Name == "" {
					secretLookupKey = types.NamespacedName{
						Name:      ExternalSecretName,
						Namespace: ExternalSecretNamespace,
					}
				}
				
Or something like that

File: e2e/suite/common/common.go [hunk 1]
```diff
@@ -57,6 +57,32 @@ func SimpleDataSync(f *framework.Framework) (string, func(*framework.TestCase))
 	}
 }
 
+// This case creates a secret with empty target name to test if it defaults to external secret name.
+func SyncWithoutTargetName(f *framework.Framework) (string, func(*framework.TestCase)) {
+	return "[common] should sync simple secrets from .Data[]", func(tc *framework.TestCase) {
```
Comment (knelasevero) [inline e2e/suite/common/common.go:62]: You should change this message that you copied from the other test

Comment (omateusp) [reply e2e/suite/common/common.go]: done

**Comment by omateusp** [comment]
Created the JSON func, should be ok to test now.

**Pull request merged**